fix: reenable debug symbols in debug builds

cyqsimon opened the pull request:
I was trying to debug #521 and was very confused when my breakpoints weren't working at all.

I mean, not having debug symbols defeats the purpose of a debug profile in the first place. So I think it's best to reenable it.

cafkafk reviewed (changes requested):
This makes sense to me, but could you e.g. `hyperfine` building with and without debug symbols so we get a sense of the scale of the difference?

cyqsimon commented:
```bash
❯ git checkout emit-debug-symbols
Switched to branch 'emit-debug-symbols'
Your branch is up to date with 'origin/emit-debug-symbols'.
❯ hyperfine --prepare 'cargo clean' 'cargo build -j4'
Benchmark 1: cargo build -j4
  Time (mean ± σ):     30.126 s ±  3.605 s    [User: 86.324 s, System: 10.159 s]
  Range (min … max):   23.004 s … 33.312 s    10 runs
❯ git checkout main
Switched to branch 'main'
Your branch is up to date with 'upstream/main'.
❯ hyperfine --prepare 'cargo clean' 'cargo build -j4'
Benchmark 1: cargo build -j4
  Time (mean ± σ):     28.213 s ±  1.516 s    [User: 79.153 s, System: 10.228 s]
  Range (min … max):   26.769 s … 31.847 s    10 runs
```

I used `-j4` to minimise the effect from other processes on my computer. There is approximately a 7% difference.

cafkafk reviewed (approved):
I guess this isn't an issue then, LGTM :+1:
